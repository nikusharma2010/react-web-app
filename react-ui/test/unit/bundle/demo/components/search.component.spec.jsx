import React from 'react';
import expect from 'expect';
import { create } from "react-test-renderer";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import SearchComponent from '../../../../../src/bundle/demo/components/search.component';
import { Link } from 'react-router-dom';

/** Check the Application Module exists**/
Enzyme.configure({ adapter: new Adapter() });

describe("Search Component", () => {
    test("it matches the snapshot", () => {
        const component = create(<BrowserRouter><SearchComponent /></BrowserRouter>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe("Test Link ", () => {
    const event = { target: { name: "employeeId", value: "1" } };
    it('Check Value', () => {
        const wrapper = mount(<BrowserRouter><SearchComponent state={{ employeeId: '' }} /></BrowserRouter>);
        wrapper.find('input').simulate('change', event);
        expect(wrapper.find(Link).props().to).toEqual("/view/1");
    });

});
