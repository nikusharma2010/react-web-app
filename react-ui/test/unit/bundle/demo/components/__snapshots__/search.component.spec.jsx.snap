// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component it matches the snapshot 1`] = `
<div
  className="styles__Row-sc-18xwbcb-2 MjgCm"
>
  <h3>
    Search By Employee Number
  </h3>
  <div
    className="styles__FormDiv-sc-18xwbcb-10 hMsRmX"
  >
    <div
      className="styles__Row-sc-18xwbcb-2 MjgCm"
    >
      <div
        className="styles__Col-sc-18xwbcb-1 cOYgjR"
      >
        <label>
          Number 
        </label>
      </div>
      <div
        className="styles__Col-sc-18xwbcb-1 cOYgjR"
      >
        <input
          className="styles__Input-sc-18xwbcb-12 jqkTFa"
          id="employeeId"
          name="employeeId"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="styles__Row-sc-18xwbcb-2 MjgCm"
    >
      <div
        className="styles__Col-sc-18xwbcb-1 cOYgjR"
      >
        <a
          href="/view/"
          id="viewEmployee"
          onClick={[Function]}
        >
          <div
            className="styles__Button-sc-18xwbcb-5 styles__ButtonPrimary-sc-18xwbcb-6 YFSZa"
          >
            View 
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
